ibm/service/dnsservices/resource_ibm_private_dns_custom_resolver_secondary_zone.go

```diff
@@ -0,0 +1,269 @@
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			pdnsInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique identifier of a service instance.",
+			},
+			pdnsResolverID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique identifier of a custom resolver.",
+			},
+			pdnsSecondaryZoneID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Secondary Zone ID",
+			},
+
+			pdnsSecondaryZoneZone: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Secondary Zone Zone",
+			},
+
+			pdnsSecondaryZoneTransferFrom: {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "Secondary Zone Zone",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+
+			pdnsSecondaryZoneEnabled: {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Secondary Zone Enabled",
+			},
+
+			pdnsSecondaryZoneDescription: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Secondary Zone Description",
+			},
+
+			pdnsSecondaryZoneCreatedOn: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Secondary Zone Creation date",
+			},
+
+			pdnsSecondaryZoneModifiedOn: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Secondary Zone Modification date",
+			},
+		},
+	}
+}
+
+func resourceIBMPrivateDNSSecondaryZoneCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).PrivateDNSClientSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	instanceID := d.Get(pdnsInstanceID).(string)
+	resolverID := d.Get(pdnsResolverID).(string)
+	description := d.Get(pdnsSecondaryZoneDescription).(string)
+	zone := d.Get(pdnsSecondaryZoneZone).(string)
+	enabled := d.Get(pdnsSecondaryZoneEnabled).(bool)
+	transferFrom := flex.ExpandStringList(d.Get(pdnsSecondaryZoneTransferFrom).([]interface{}))
+
+	createSecondaryZoneOptions := sess.NewCreateSecondaryZoneOptions(instanceID, resolverID)
+
+	createSecondaryZoneOptions.SetZone(zone)
+	createSecondaryZoneOptions.SetDescription(description)
+	createSecondaryZoneOptions.SetEnabled(enabled)
+	createSecondaryZoneOptions.SetTransferFrom(transferFrom)
+
+	mk := "private_dns_secondary_zone_" + instanceID + resolverID
+	conns.IbmMutexKV.Lock(mk)
+	defer conns.IbmMutexKV.Unlock(mk)
+
+	resource, response, err := sess.CreateSecondaryZone(createSecondaryZoneOptions)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("[ERROR] Error creating pdns secondary zone:%s\n%s", err, response))
```

Comment:
Error message should not use internal code name [pdns] instead to be using DNS services.